core/tests/protractor_utils/CreatorDashboardPage.js

```diff
@@ -44,225 +48,219 @@ var CreatorDashboardPage = function() {
-      by.css('.protractor-test-exploration-dashboard-card'));
-    return allExplorationDashboardCard.then(function(tiles) {
-      return tiles.filter(function(tile) {
-        return tile.getText().then(function(text) {
-          // Tile text contains title, possibly followed by newline and text
-          return (
-            text.startsWith(explorationTitle + '\n') ||
-            text === explorationTitle
-          );
-        });
-      });
+  // Returns all exploration card elements with the given name.
+  var _getExplorationElements = async function(explorationTitle) {
+    await waitFor.visibilityOf(explorationDashboardCard);
+    return await allExplorationCards.filter(async function(tile) {
+      var text = await tile.getText();
+      // Tile text contains title, possibly followed by newline and text
+      return (
+        text.startsWith(explorationTitle + '\n') ||
+        text === explorationTitle
+      );
     });
   };
 
-  this.get = function() {
-    browser.get(CREATOR_DASHBOARD_URL);
-    return waitFor.pageToFullyLoad();
+  this.get = async function() {
+    await browser.get(CREATOR_DASHBOARD_URL);
+    await waitFor.pageToFullyLoad();
   };
 
-  this.getNumberOfFeedbackMessages = function() {
-    return explorationFeedbackCount.getText().then(function(text) {
-      return parseInt(text);
-    });
+  this.getNumberOfFeedbackMessages = async function() {
+    var feedbackCount = await explorationFeedbackCount.getText();
+    return parseInt(feedbackCount);
   };
 
-  this.navigateToExplorationEditor = function() {
-    waitFor.elementToBeClickable(explorationDashboardCard);
-    explorationDashboardCard.click();
-    waitFor.pageToFullyLoad();
+  this.navigateToExplorationEditor = async function() {
+    await waitFor.elementToBeClickable(explorationDashboardCard);
+    await explorationDashboardCard.click();
+    await waitFor.pageToFullyLoad();
   };
 
-  this.clickCreateActivityButton = function() {
-    waitFor.elementToBeClickable(
+  this.clickCreateActivityButton = async function() {
+    await waitFor.elementToBeClickable(
       createActivityButton,
       'Create Activity button takes too long to be clickable');
-    createActivityButton.click();
-    waitFor.pageToFullyLoad();
+    await createActivityButton.click();
+    await waitFor.pageToFullyLoad();
   };
 
-  this.clickCreateCollectionButton = function() {
-    waitFor.visibilityOf(
+  this.clickCreateCollectionButton = async function() {
+    await waitFor.visibilityOf(
       activityCreationModal, 'Activity Creation modal is not visible');
-    waitFor.elementToBeClickable(
+    await waitFor.elementToBeClickable(
       createCollectionButton,
       'Create Collection button takes too long to be clickable');
-    createCollectionButton.click();
-    waitFor.pageToFullyLoad();
+    await createCollectionButton.click();
+    await waitFor.pageToFullyLoad();
+    await waitFor.visibilityOf(collectionEditorContainer);
```

Comment:
Are these redundant?

Reply:
Not necessarily. Wait for page to load just checks if the navigation has completed. The visibility check could be to check if the specific element is available for further operation.